### PR TITLE
Spotify Reader fix user_id key

### PR DESCRIPTION
**Pull request by amCap1712**
In #1670 , I had used the wrong key in user dict used by spotify reader. Fixing that here and adding a test to ensure that listens are submitted correctly by spotify reader.

### CODE REVIEW

File: listenbrainz/spotify_updater/spotify_read_listens.py [hunk 1]
```diff
@@ -232,6 +232,8 @@ def submit_listens_to_listenbrainz(user: Dict, listens: List, listen_type=LISTEN
         listen_type: the type of listen (single, import, playing_now)
     """
     username = user['musicbrainz_id']
+    # insert payload looks for 'id' key whereas the user dict here has it in 'user_id' key
+    user['id'] = user['user_id']
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply listenbrainz/spotify_updater/spotify_read_listens.py]: Fixed.

File: listenbrainz/spotify_updater/spotify_read_listens.py [hunk 2]
```diff
@@ -232,11 +234,13 @@ def submit_listens_to_listenbrainz(user: Dict, listens: List, listen_type=LISTEN
         listen_type: the type of listen (single, import, playing_now)
     """
     username = user['musicbrainz_id']
+    # insert payload looks for 'id' key whereas the user dict here has it in 'user_id' key
```
Comment (alastair) [inline listenbrainz/spotify_updater/spotify_read_listens.py:237]: Is this comment right? in the next line you use `user['user_id']` but the comment says `'id'`. id is used in the api endpoint.

Comment (amCap1712) [reply listenbrainz/spotify_updater/spotify_read_listens.py:237]: Yeah, the comment is accurate for the old code that did user['id'] = user['user_id'] but no longer needed that we now have the type. Will remove the comment.

Comment (amCap1712) [reply listenbrainz/spotify_updater/spotify_read_listens.py]: Fixed.

**Review by alastair — APPROVED** [review]
Failing test looks unrelated. Looks good to me!

**Pull request merged**